lsp-methods.el

```diff
@@ -1421,9 +1421,25 @@ If title is nil, return the name for the command handler."
   "Ask the server to format this document."
   (interactive)
   (let ((edits (lsp--send-request (lsp--make-request
-                                    "textDocument/formatting"
-                                    (lsp--make-document-formatting-params)))))
-    (save-excursion (lsp--apply-text-edits edits))))
+                                   "textDocument/formatting"
+                                   (lsp--make-document-formatting-params)))))
+    (if (fboundp 'replace-buffer-contents)
+        (let ((content (buffer-string))
```

Comment:
Prefer `buffer-substring-no-properties` to avoid copying the properties around.

Even better, use `insert-buffer-substring-no-properties` below to avoid creating a temporary string.

Reply:
@phst Yes `insert-buffer-substring-no-properties` seems better. And thanks for your work on `replace-buffer-contents` :)